From 615160ff2f1e6f0f0ee324f7442b0068552d0068 Mon Sep 17 00:00:00 2001
From: Maksim Ivanov <emaxx@chromium.org>
Date: Wed, 18 May 2022 21:38:13 +0200
Subject: [PATCH] Fix shutdown memory leak of sReadersContexts

RFCleanupReaders() should always free memory of sReadersContexts
(allocated in RFAllocateReaderSpace()), not only when the reader was
active.

This was only a minor, shutdown-only, issue.
---
 src/readerfactory.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/readerfactory.c b/src/readerfactory.c
index 50ff96d1..d6041f30 100644
--- a/src/readerfactory.c
+++ b/src/readerfactory.c
@@ -1396,11 +1396,10 @@ void RFCleanupReaders(void)
 
 			if (rv != SCARD_S_SUCCESS)
 				Log2(PCSC_LOG_ERROR, "RFRemoveReader error: 0x%08lX", rv);
-
-			free(sReadersContexts[i]);
-
-			sReadersContexts[i] = NULL;
 		}
+
+		free(sReadersContexts[i]);
+		sReadersContexts[i] = NULL;
 	}
 
 #ifdef USE_SERIAL
